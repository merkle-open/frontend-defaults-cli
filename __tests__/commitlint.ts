/// <reference types="@types/jest" />
import { apiIt } from './shared';

describe('commitlint', () => {
	it('default', async () =>
		await apiIt('commitlint', {
			commitlint: true,
		}));
});
