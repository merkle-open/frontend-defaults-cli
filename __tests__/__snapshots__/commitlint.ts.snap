// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commitlint default 1`] = `
"{
  \\"options\\": {
    \\"ts\\": false,
    \\"tslint\\": false,
    \\"es\\": false,
    \\"eslint\\": false,
    \\"editorconfig\\": false,
    \\"prettier\\": false,
    \\"stylelint\\": false,
    \\"gitignore\\": false,
    \\"npmrc\\": false,
    \\"readme\\": false,
    \\"githooks\\": true,
    \\"commitlint\\": true,
    \\"nodeVersion\\": false,
    \\"webpack\\": false,
    \\"build\\": false,
    \\"install\\": false,
    \\"force\\": true,
    \\"dryRun\\": false,
    \\"mode\\": \\"api\\"
  },
  \\"mergedFiles\\": {
    \\"package.json\\": {
      \\"override\\": false,
      \\"data\\": {
        \\"scripts\\": {
          \\"cz\\": \\"git-cz\\"
        },
        \\"husky\\": {
          \\"hooks\\": {
            \\"commit-msg\\": \\"commitlint -E HUSKY_GIT_PARAMS\\"
          }
        },
        \\"commitlint\\": {
          \\"extends\\": [
            \\"@namics/commitlint-conventional-changelog\\"
          ]
        },
        \\"config\\": {
          \\"commitizen\\": {
            \\"path\\": \\"@namics/cz-conventional-changelog\\"
          }
        },
        \\"devDependencies\\": {
          \\"@commitlint/cli\\": \\"7.5.2\\",
          \\"@namics/commitlint-conventional-changelog\\": \\"1.0.1\\",
          \\"@namics/cz-conventional-changelog\\": \\"1.0.2\\",
          \\"commitizen\\": \\"3.0.7\\",
          \\"husky\\": \\"1.3.1\\",
          \\"lint-staged\\": \\"8.1.4\\"
        }
      }
    },
    \\".huskyrc\\": {
      \\"override\\": false,
      \\"data\\": \\"# ~/.huskyrc\\\\n# Note: it will run in /bin/sh shell\\\\nif command -v nodenv >/dev/null 2>&1; then\\\\n  eval \\\\\\"$(nodenv init -)\\\\\\"\\\\nfi\\"
    }
  }
}"
`;

exports[`commitlint default 2`] = `
"# ~/.huskyrc
# Note: it will run in /bin/sh shell
if command -v nodenv >/dev/null 2>&1; then
  eval \\"$(nodenv init -)\\"
fi"
`;

exports[`commitlint default 3`] = `
"{
  \\"scripts\\": {
    \\"cz\\": \\"git-cz\\"
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"commit-msg\\": \\"commitlint -E HUSKY_GIT_PARAMS\\"
    }
  },
  \\"commitlint\\": {
    \\"extends\\": [
      \\"@namics/commitlint-conventional-changelog\\"
    ]
  },
  \\"config\\": {
    \\"commitizen\\": {
      \\"path\\": \\"@namics/cz-conventional-changelog\\"
    }
  },
  \\"devDependencies\\": {
    \\"@commitlint/cli\\": \\"7.5.2\\",
    \\"@namics/commitlint-conventional-changelog\\": \\"1.0.1\\",
    \\"@namics/cz-conventional-changelog\\": \\"1.0.2\\",
    \\"commitizen\\": \\"3.0.7\\",
    \\"husky\\": \\"1.3.1\\",
    \\"lint-staged\\": \\"8.1.4\\"
  }
}"
`;
